patches/net/minecraft/core/NonNullList.java.patch

```diff
@@ -0,0 +1,30 @@
+--- a/net/minecraft/core/NonNullList.java
++++ b/net/minecraft/core/NonNullList.java
+@@ -9,6 +_,27 @@
+ import org.apache.commons.lang3.Validate;
+ 
+ public class NonNullList<E> extends AbstractList<E> {
++
++    /**
++     * Neo: utility method to construct a Codec for a NonNullList
++     * @param entryCodec the codec to use for the elements
++     * @param <E> the element type
++     * @return a codec that encodes as a list, and decodes into NonNullList
++     */
++    public static <E> com.mojang.serialization.Codec<NonNullList<E>> codecOf(com.mojang.serialization.Codec<E> entryCodec) {
++        return entryCodec.listOf().xmap(NonNullList::copyOf, java.util.function.Function.identity());
++    }
++
++    /**
++     * Neo: utility method to construct an immutable NonNullList from a given collection
++     * @param entries the collection to make a copy of
++     * @param <E> the type of the elements in the list
++     * @return a new immutable NonNullList
++     */
++    public static <E> NonNullList<E> copyOf(java.util.Collection<? extends E> entries) {
```

Comment:
On a second thought, I'd document the NPE the same way `List#copyOf` does:
>      * @throws NullPointerException if coll is null, or if it contains any nulls

Reply:
OH! I didn't realize List.copyOf disallows nulls. HAH
I made it null-safe accidentally.